CCPAPI/sip/CMakeLists.txt

```diff
@@ -0,0 +1,149 @@
+
+##########################################################################
+#                                                                        #
+#                               SIP                                      #
+#                                                                        #
+#  This program is free software; you can redistribute it and/or modify  #
+#  it under the terms of the GNU Library General Public License as       #
+#  published by the Free Software Foundation; version 2 or later of the  #
+#  License.                                                              #
+#                                                                        #
+#  This program is distributed in the hope that it will be useful,       #
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of        #
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the          #
+#  GNU General Public License for more details.                          #
+#                                                                        #
+#          Copyright 2020 Paul RASCLE www.openfields.fr                  #
+#                                                                        #
+##########################################################################
+
+message(STATUS "SIP processing")
+
+# --- Python detection: if several versions installed, select one with all the required packages installed
+
+set(PYTHON_PREFERED_VERSION "3.6" CACHE STRING "Select a Python version: it should be installed with all packages required for CloudCompare (pyqt, sip, numpy)")
+set_property(CACHE PYTHON_PREFERED_VERSION PROPERTY STRINGS "3.6" "3.7" "3.8")
+message(STATUS "python prefered version: " ${PYTHON_PREFERED_VERSION})
+set(Python_ADDITIONAL_VERSIONS ${PYTHON_PREFERED_VERSION}) # priority to this version 
+find_package(PythonInterp)
+find_package(PythonLibs)
+message(STATUS "python version: " ${PYTHON_VERSION_STRING})
+message(STATUS "check match for ${PYTHON_PREFERED_VERSION}.x")
+if(NOT ${PYTHON_VERSION_STRING} MATCHES "${PYTHON_PREFERED_VERSION}\\..")
+    message(SEND_ERROR "no Python found matching ${PYTHON_PREFERED_VERSION}.x")
+else()
+    message(STATUS "Python found ${PYTHON_VERSION_STRING} matches prefered version ${PYTHON_PREFERED_VERSION}.x")
+    message(STATUS "PYTHON_LIBRARIES:" ${PYTHON_LIBRARIES})
+endif()
+
+# --- PyQt Sip detection
+
+find_package(PYQTSIP REQUIRED)  # should come after Python
+
+# --- ----------------------------------------------------------
+# --- target PyQt sip Python library: Python module cloudCompare
+
+project( cloudCompare VERSION 0.1 )
```

Comment:
Is this name going to conflict with the main CloudCompare project (in qCC)? (Not sure of build order/inclusion/etc.)

Maybe PyCloudCompareLib or PyCloudCompare?